pkg/reconciler/cronjobsource/cronjobsource.go

```diff
@@ -122,51 +123,67 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 	return err
 }
 
-func (r *Reconciler) reconcile(ctx context.Context, cronjob *v1alpha1.CronJobSource) error {
+func (r *Reconciler) reconcile(ctx context.Context, source *v1alpha1.CronJobSource) error {
 	// This Source attempts to reconcile three things.
 	// 1. Determine the sink's URI.
 	//     - Nothing to delete.
 	// 2. Create a receive adapter in the form of a Deployment.
 	//     - Will be garbage collected by K8s when this CronJobSource is deleted.
 	// 3. Create the EventType that it can emit.
 	//     - Will be garbage collected by K8s when this CronJobSource is deleted.
-	cronjob.Status.ObservedGeneration = cronjob.Generation
-
-	cronjob.Status.InitializeConditions()
-
-	sinkObjRef := cronjob.Spec.Sink
-	if sinkObjRef.Namespace == "" {
-		sinkObjRef.Namespace = cronjob.Namespace
+	source.Status.ObservedGeneration = source.Generation
+
+	source.Status.InitializeConditions()
+
+	dest := source.Spec.Sink.DeepCopy()
```

Comment:
Do we need to check if Sink is nil? apiserversource has https://github.com/knative/eventing/pull/2089/files#diff-d99b024fa0cd240e5a9ba150e807531aR148. Without this I think we'll get a panic trying to call DeepCopy, right?

Reply:
It should not be: https://github.com/knative/eventing/blob/master/pkg/apis/sources/v1alpha1/cron_job_validation.go#L39-L40

Reply:
```go
// We should never get here
```